File: 프로그래머스/2/42586. 기능개발/기능개발.py
from collections import deque
def solution(progresses, speeds):
    answer = []
    q=deque()
    for i in range(len(progresses)):
        q.append([progresses[i],speeds[i]])
    while q:
        count=0
        for i in range(len(q)):
            q[i][0]+=q[i][1]
        while(q and q[0][0]>=100):
            q.popleft()
            count+=1
        if count>0:
            answer.append(count)



    return answer

File: 프로그래머스/2/42586. 기능개발/기능개발.js
function solution(progresses, speeds) {
    var answer = [];
    let queue=[]
    
    for(let i=0; i<progresses.length; i++){
        queue.push([progresses[i],speeds[i]]) // 큐에 값 추가 
    }
    //console.log(queue)
    //console.log(queue[0], queue[0][0])
    
    
    while (queue.length > 0 ){ // queue가 비어질때까지 
        let count=0;
        flag = false;
        for(let i=0; i<queue.length;i++){
            queue[i][0]+=queue[i][1] // 매일 진행하기 
            }
        
        while(queue.length > 0 && queue[0][0]>=100){ // 배포하기
            queue.shift()
            count+=1
        }
        
        if (count>0){
            answer.push(count)
        }

            
        }
        
    return answer;
        
    }
    
    
    
    
    
    
    
    
